entry.sh

```diff
@@ -30,10 +30,22 @@ fi
 # NOTE: this won't be necessary once the supervisor can update
 # itself, as using the label io.balena.features.journal-logs will
 # achieve the same objective
-if [ -d /mnt/root/run/log/journal ]; then
+if { [ ! -d /run/log/journal ] || [ -L /run/log/journal ]; } && [ -s /mnt/root/etc/machine-id ]; then
```

Comment:
```suggestion
if { [ ! -d /run/log/journal ] || [ ! -L /run/log/journal ]; } && [ -s /mnt/root/etc/machine-id ]; then
```
Not a symlink --> forgot the `!`

Reply:
Otherwise LGTM!

Reply:
Sorry, my comment in the line below is wrong. `-d` returns true even if the file is a symlink. If the directory exists, then only if the file is a symlink we want to enter the `if` clause. I'll fix the comment

Reply:
No, my comment  comment in the following line is wrong. If the directory exists, then we only want to enter the `if` section if the file is a symlink (`-d` returns true

Reply:
I updated the comment